data2rdf/pipelines/annotation_pipeline.py

```diff
@@ -0,0 +1,186 @@
+        None,
+        description="Filepath or rdflib-object for a Graph with extra triples for the resulting pipeline graph.",
+    )
+
+    model_config = ConfigDict(
+        arbitrary_types_allowed=True, use_enum_values=True
+    )
+
+    @field_validator("config")
+    @classmethod
+    def validate_config(cls, value: Union[Dict[str, Any], Config]) -> Config:
+        """Validate configuration"""
+        if isinstance(value, dict):
+            value = Config(**value)
+        return value
+
+    @field_validator("extra_triples")
+    @classmethod
+    def validate_extra_triples(
+        cls, value: Optional[Union[str, Graph]], info: ValidationInfo
+    ) -> Graph:
+        """Validate extra triples."""
+        config = info.data.get("config")
+        if isinstance(value, str):
+            with open(value, encoding=config.encoding) as file:
+                extra_triples = file.read()
+        elif isinstance(value, Graph):
+            extra_triples = value.serialize()
+        else:
+            raise TypeError(
+                f"`extra_triples` must be of type {str} or {Graph}, not {type(value)}."
+            )
+        extra_triples = extra_triples.replace(
+            config.namespace_placeholder, str(config.base_iri)
+        )
+        graph = Graph(identifier=config.graph_identifier)
+        graph.parse(data=extra_triples)
+        return value
+
+    @model_validator(mode="after")
+    @classmethod
+    def run_pipeline(cls, self: "AnnotationPipeline") -> "AnnotationPipeline":
+        """Run pipeline."""
+        self.parser = self.parser(
+            raw_data=self.raw_data,
+            mapping=self.mapping,
+            config=self.config,
+            **self.parser_args,
+        )
+
+        return self
+
+    @property
+    def json_ld(cls) -> Dict[str, Any]:
+        """Return dict of json-ld for graph"""
+        return {
+            "@context": {
+                "fileid": make_prefix(cls.config),
+                "csvw": "http://www.w3.org/ns/csvw#",
+                "rdfs": "http://www.w3.org/2000/01/rdf-schema#",
+                "dcat": "http://www.w3.org/ns/dcat#",
+                "xsd": "http://www.w3.org/2001/XMLSchema#",
+                "dcterms": "http://purl.org/dc/terms/",
+                "qudt": "http://qudt.org/schema/qudt/",
+                "csvw": "http://www.w3.org/ns/csvw#",
+                "foaf": "http://xmlns.com/foaf/spec/",
+            },
+            "@id": "fileid:dataset",
+            "@type": "dcat:Dataset",
+            "dcat:distribution": {
+                "@type": "dcat:Distribution",
+                "dcat:mediaType": {
+                    "@type": "xsd:anyURI",
+                    "@value": cls.parser.media_type,
+                },
+                "dcat:accessURL": {
+                    "@type": "xsd:anyURI",
+                    "@value": str(cls.config.data_download_uri),
+                },
+            },
+            "dcterms:hasPart": cls.parser.json_ld,
```

Comment:
Is this where the RDF data (which was generated by one of the parsers) being added to the final rdf file?

Reply:
Exactly!